Lib/importlib/_bootstrap.py

```diff
@@ -1395,6 +1395,10 @@ def _gcd_import(name, package=None, level=0):
     _sanity_check(name, package, level)
     if level > 0:
         name = _resolve_name(name, package, level)
+    module = sys.modules.get(name, _NEEDS_LOADING)
+    if (module is _NEEDS_LOADING or
+        getattr(getattr(module, "__spec__", None), "_initializing", False)):
```

Comment:
I don't love that I am repeating this code that occurs in `_find_and_load`, but if I don't we will audit imports that have already occurred. Moving the audit into `_find_and_load` results in double logging of imports. Perhaps I could do a larger refactor, but wanted to avoid that unless others think it's best.

Reply:
Why not move the audit to `_find_and_load_unlocked()` and remove it from `import.c`?

Reply:
Mostly because I get nervous to make changes to the C code :) Updated it with the new location